src/ripple/app/main/Application.cpp

```diff
@@ -862,39 +861,52 @@ class ApplicationImp
 
     //------------------------------------------------------------------------------
 
-    void onDeadlineTimer (DeadlineTimer& timer) override
+    void setSweepTimer ()
     {
-        if (timer == m_entropyTimer)
-        {
-            crypto_prng().mix_entropy ();
-            return;
-        }
-
-        if (timer == m_sweepTimer)
-        {
-            // VFALCO TODO Move all this into doSweep
-
-            if (! config_->standalone())
+        sweepTimer_.expires_from_now (
+            std::chrono::seconds {config_->getSize (siSweepInterval)});
+        sweepTimer_.async_wait ([this] (boost::system::error_code const& e)
             {
-                boost::filesystem::space_info space =
-                        boost::filesystem::space (config_->legacy ("database_path"));
-
-                // VFALCO TODO Give this magic constant a name and move it into a well documented header
-                //
-                if (space.available < (512 * 1024 * 1024))
+            if ((e.value() == boost::system::errc::success) &&
+                (! m_jobQueue->isStopped()))
                 {
-                    JLOG(m_journal.fatal())
-                        << "Remaining free disk space is less than 512MB";
-                    signalStop ();
+                    m_jobQueue->addJob(
+                        jtSWEEP, "sweep", [this] (Job&) { doSweep(); });
                 }
-            }
+            });
+    }
 
-            m_jobQueue->addJob(jtSWEEP, "sweep", [this] (Job&) { doSweep(); });
-        }
+    void setEntropyTimer ()
+    {
+        using namespace std::chrono_literals;
+        entropyTimer_.expires_from_now (5min);
+        entropyTimer_.async_wait ([this] (boost::system::error_code const& e)
+            {
+                if (e.value() == boost::system::errc::success)
```

Comment:
I don't think we *need* to terminate, although we could and it's probably safer if we do.

Longer term, I'd like something more elegant that does away with the need for a timer (namely, Fortuna) that can just be fed entropy directly.